frontend/src/metabase/visualizations/components/settings/ChartSettingHideButton.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useCallback } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { t } from "ttag";
+import cloneDeep from "lodash.clonedeep";
+import _ from "underscore";
+
+import Button from "metabase/core/components/Button";
+import { updateCardVisualizationSettings } from "metabase/query_builder/actions";
+import type { Column } from "metabase-types/types/Dataset";
+import type { State } from "metabase-types/store";
+
+import { TableColumnOrderSetting } from "metabase-types/api";
+import { normalizeFieldRef } from "metabase-lib/queries/utils/dataset";
+
+interface ChartSettingHideButtonProps {
+  column: Column;
+  columns: Array<Column>;
+  onClose?: () => void;
+}
+
+export function ChartSettingHideButton({
+  column,
+  columns,
+  onClose,
+}: ChartSettingHideButtonProps) {
+  const hideColumn = useHideColumn(column, columns);
+
+  return (
+    <Button
+      onlyText
+      icon="eye_crossed_out"
+      onClick={() => {
+        hideColumn();
+        onClose?.();
+      }}
+    >
+      {t`Hide this column`}
+    </Button>
+  );
+}
+
+function useHideColumn(column: Column, columns: Array<Column>) {
+  const dispatch = useDispatch();
+  const visualizationSettings = useSelector(
+    (state: State) => state.qb.card?.visualization_settings,
+  );
+
+  return useCallback(() => {
+    let visSettingsClone = cloneDeep(visualizationSettings);
+    if (visSettingsClone === undefined) {
+      visSettingsClone = {};
+    }
+    if (visSettingsClone["table.columns"] === undefined) {
+      visSettingsClone["table.columns"] = getDefaultTableColumns(columns);
+    }
+
+    const tableColIndex = findTableColumnIndex(
+      visSettingsClone?.["table.columns"],
+      column,
+    );
+    if (tableColIndex === -1) {
+      return;
+    }
+
+    const tableCol = visSettingsClone["table.columns"][tableColIndex];
+    visSettingsClone["table.columns"][tableColIndex] = {
+      ...tableCol,
+      enabled: false,
+    };
+    // @ts-expect-error thunk
+    dispatch(updateCardVisualizationSettings(visSettingsClone));
+  }, [dispatch, visualizationSettings, column, columns]);
+}
+
+export function getDefaultTableColumns(columns: Array<Column>) {
```

Comment:
Since this is used both here and in `Table`, a better place to define it would have been in `metabase/visualizations/lib/settings/utils`. However, there is already a `utils.js` file there written in vanilla javascript, but I needed this to be written in typescript for usage in the `useHideColumn` hook above.